src/app/campaign-list/campaign-list.service.ts

```diff
@@ -37,34 +55,37 @@ export interface Campaign {
   repName: string;
   notes: string;
   flights?: Flight[];
+  loading?: boolean;
 }
 
 @Injectable()
 export class CampaignListService {
-  params = {page: 1, per: 12};
-  facets: {
-    advertiser: Facet[],
-    podcast: Facet[],
-    status: Facet[],
-    type: Facet[]
-  };
+  params: CampaignParams = {page: 1, per: 12};
+  facets: Facets;
   total: number;
   count: number;
   error: Error;
-  loading = false;
-  flightsLoaded: number;
   // tslint:disable-next-line: variable-name
   private _campaigns = new BehaviorSubject<{[id: number]: Campaign}>({});
```

Comment:
this._campaigns holds all campaigns that have been requested, so if we did decide to do something with using this application state as a cache, it's here

Reply:
+1